driver-core/src/main/com/mongodb/internal/operation/QueryBatchCursor.java

```diff
@@ -579,9 +584,10 @@ private void killServerCursor(final MongoNamespace namespace, final ServerCursor
             long cursorId = serverCursor.getId();
             if (serverIsAtLeastVersionThreeDotTwo(connection.getDescription())) {
                 connection.command(namespace.getDatabaseName(), asKillCursorsCommandDocument(namespace, serverCursor),
-                        NO_OP_FIELD_NAME_VALIDATOR, ReadPreference.primary(), new BsonDocumentCodec(), sessionContext, serverApi);
+                        NO_OP_FIELD_NAME_VALIDATOR, ReadPreference.primary(), new BsonDocumentCodec(), sessionContext, serverApi,
+                        requestContext());
             } else {
-                connection.killCursor(namespace, singletonList(cursorId));
+                connection.killCursor(namespace, singletonList(cursorId), requestContext());
```

Comment:
May we please pass `RequestContext` to `killServerCursor` instead of getting it from inside the method? It was intended that the methods `killServerCursor` and `asKillCursorsCommandDocument` are static, but Java has an unhelpful limitation (I understand why the language authors might have decided to do this, but I do not think it is justifiable) that prevents declaring static methods in inner classes.

Reply:
Yes, it is not. The artificial and unhelpful limitation of the Java syntax is to blame. There are two ways this can be worked around:

1. Make the `ResourceManager` class static. This is how it was originally in the PR that added it, but we decided to make it an inner class during the review process to shorten the code.
2. Move the `killServerCursor`, `asKillCursorsCommandDocument` methods out of `ResourceManager` to `QueryBatchCursor` and make them static. This defeats the idea of consolidating the logic related to managing both client and server resources in `ResourceManager`.

Maybe `ResourceManager` should have stayed static 🤷‍♂️. All approaches are not good.